### ISSUE TITLE
CLI Chat wont start

### ISSUE THREAD

**Issue by press74** [pinecone-io/canopy#166]
### Is this a new bug?

- [X] I believe this is a new bug
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Start canopy, switch to new terminal to run "canopy chat" and it says that canopy isnt running

### Expected Behavior

Canopy chat starts

### Steps To Reproduce

Win environment
Occurs inside and outside virtual environment

### Relevant log output

```shell
Starting Canopy server on 0.0.0.0:8000
INFO:     Started server process [6148]
INFO:     Waiting for application startup.
2023-11-09 20:47:46,231 - MainProcess - canopy_server.app [INFO    ]:  Did not find config file. Initializing engines with default configuration
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)

At same time:
Error: Canopy server is not running on http://0.0.0.0:8000.
please run `canopy start`
```

### Environment

```markdown
- Win:
- python3
```

### Additional Context

I am not a coder, just hobbyist, so likely I am just missing something

**Comment by danielcmm** [comment]
Try canopy start --host 127.0.0.1

**Comment by igiloh-pinecone** [comment]
Hi @press74, thanks for reporting this problem!
It appears that `http:0.0.0.0:8000` indeed doesn't resolve correctly on windows. Can you please try
```
canopy chat --chat-server-url http://localhost:8000
```
or maybe 
```
canopy chat --chat-server-url http://127.0.0.1:8000
```
like @danielcmm has suggested?

We will fix it in the next version of Canopy, so it will automatically work on Windows.

**Comment by press74** [comment]
thank you so much for the quick response. Chat is now connecting, but I am getting the error below. is there a random \ in the code somewhere when using CLI in the terminal? Works fine using the localhost in the browser as you can see from the last line.

INFO:     127.0.0.1:62130 - "GET /health HTTP/1.1" 200 OK
INFO:     127.0.0.1:62142 - "POST /%5Ccontext/chat/completions HTTP/1.1" 404 Not Found
INFO:     127.0.0.1:62271 - "GET /docs HTTP/1.1" 200 OK
INFO:     127.0.0.1:62272 - "GET /docs HTTP/1.1" 200 OK
INFO:     127.0.0.1:62297 - "GET /docs HTTP/1.1" 200 OK
INFO:     127.0.0.1:62297 - "GET /openapi.json HTTP/1.1" 200 OK
INFO:     127.0.0.1:62298 - "POST /context/chat/completions HTTP/1.1" 200 OK

**Comment by ss-syndev** [comment]
> Hi @press74, thanks for reporting this problem! It appears that `http:0.0.0.0:8000` indeed doesn't resolve correctly on windows. Can you please try
> 
> ```
> canopy chat --chat-server-url http://localhost:8000
> ```
> 
> or maybe
> 
> ```
> canopy chat --chat-server-url http://127.0.0.1:8000
> ```
> 
> like @danielcmm has suggested?
> 
> We will fix it in the next version of Canopy, so it will automatically work on Windows.

I've experienced this issue as well and wanted to mention the error message when trying to run 'canopy stop' while experiencing the issues detailed above. 

Traceback (most recent call last):
  File "C:\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Python39\Scripts\canopy.exe\__main__.py", line 7, in <module>
  File "C:\Python39\lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "C:\Python39\lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "C:\Python39\lib\site-packages\click\core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Python39\Python39\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Python39\lib\site-packages\click\core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "C:\WorkStuffs\python\Python39\Python39\lib\site-packages\canopy_cli\cli.py", line 581, in stop
    res = subprocess.run(["pgrep", "-f", "gunicorn canopy_server.app:app"],
  File "C:\Python39\lib\subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Python39\lib\subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Python39\lib\subprocess.py", line 1420, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

If it's unrelated then there may be an issue with using pgrep on Win.

**Comment by igiloh-pinecone** [comment]
> thank you so much for the quick response. Chat is now connecting, but I am getting the error below. is there a random \ in the code somewhere when using CLI in the terminal? Works fine using the localhost in the browser as you can see from the last line.
> 
> INFO: 127.0.0.1:62130 - "GET /health HTTP/1.1" 200 OK INFO: 127.0.0.1:62142 - "POST /%5Ccontext/chat/completions HTTP/1.1" 404 Not Found INFO: 127.0.0.1:62271 - "GET /docs HTTP/1.1" 200 OK INFO: 127.0.0.1:62272 - "GET /docs HTTP/1.1" 200 OK INFO: 127.0.0.1:62297 - "GET /docs HTTP/1.1" 200 OK INFO: 127.0.0.1:62297 - "GET /openapi.json HTTP/1.1" 200 OK INFO: 127.0.0.1:62298 - "POST /context/chat/completions HTTP/1.1" 200 OK

@press74, thanks for the feedback!  
You are right, I had a mistake in my previous comment, it was missing a trailing `/` (edited now for future readers).  
Please try this command, it should work:
```
canopy chat --chat-server-url http://localhost:8000/
```
I just tested on Windows10 and verified its working.  

We will be releasing an official fix very soon.

**Comment by igiloh-pinecone** [comment]
> @igiloh-pinecone
> 
> I've experienced this issue as well and wanted to mention the error message when trying to run 'canopy stop' while experiencing the issues detailed above.


@ss-syndev , thank you very much for reporting this!
That's actually a separate issue, happening specifically for `canopy stop` on windows.  
Please free to open a separate issue to track that bug. We will publish a fix for both issues ASAP.

**Comment by igiloh-pinecone** [comment]
Thank you @press74 @ss-syndev for reporting these problems!  
Both were fixed in the latest release (V0.1.4) which is already available.
